namespaces/mson-namespace.md

```diff
@@ -0,0 +1,497 @@
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "member",
+            "content": {
+                "key": {
+                    "element": "string",
+                    "content": "id"
+                },
+                "value": {
+                    "element": "number",
+                    "attributes": {
+                        "default": 0
+                    },
+                    "content": null
+                }
+            }
+        }
+    ]
+}
+```
+
+### One Of
+
+#### MSON
+
+```
+- city
+- One Of
+    - state
+    - province
+```
+
+#### MSON DOM
+
+TODO:
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "member",
+            "content": {
+                "key": {
+                    "element": "string",
+                    "content": "city"
+                }
+            }
+        },
+        {
+            "element": "select",
+            "content": [
+                {
+                    "element": "option",
+                    "content": [
+                        {
+                            "element": "member",
+                            "content": {
+                                "key": {
+                                    "element": "string",
+                                    "content": "state"
+                                }
+                            }
+                        }
+                    ]
+                },
+                {
+                    "element": "option",
+                    "content": [
+                        {
+                            "element": "member",
+                            "content": {
+                                "key": {
+                                    "element": "string",
+                                    "content": "province"
+                                }
+                            }
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+### Mixin
+
+#### MSON
+
+```
+- id
+- Include User
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "extend",
+    "content": [
+        {
+            "element": "object",
+            "content": [
+                {
+                    "element": "member",
+                    "content": {
+                        "key": {
+                            "element": "string",
+                            "content": "id"
+                        }
+                    }
+                }
+            ]
+        }
+        {
+            "element": "ref",
+            "content": "User"
+        }
+    ]
+}
+```
+
+### Named Type
+
+#### MSON
+
+```
+# Address (object)
+
+Description is here! Properties to follow.
+
+## Properties
+
+- street
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "meta": {
+        "id": "Address",
+        "title": "Address",
+        "description": "Description is here! Properties to follow."
```

Comment:
So I am just wondering whether we are not overcomplicating things here – trying to be smart about future. I would, for now, get rid of description in MSON element completely, merging it all into `description` in `meta` – that is the super convenient and unified way for consumers.

If, for some reason, we will need separate block description(s – imagine there is more block descriptions separated by for example properties) later we can add them as member (content) elements. But for now I would keep it simple stupid.